src/iface/ethernet.rs

```diff
@@ -23,7 +23,8 @@ pub struct Interface<'a, 'b, 'c, DeviceT: Device + 'a> {
     device:         Managed<'a, DeviceT>,
     arp_cache:      Managed<'b, ArpCache>,
     hardware_addr:  EthernetAddress,
-    protocol_addrs: ManagedSlice<'c, IpAddress>,
+    protocol_addrs: ManagedSlice<'c, IpCidr>,
+    default_gw:     Option<IpAddress>,
```

Comment:
Nit: the phrase "default gateway" is a remnant of the times where a single LAN segment would conceivably have more than one gateway. At this point it's an anachronism (like classful networks!), let's call this just "gateway". Also applies elsewhere in this PR.